src/term/mod.rs

```diff
@@ -418,15 +418,12 @@ impl<'a> Iterator for RenderableCellsIter<'a> {
                 let selected =
                     self.selection.as_ref().map(|range| range.contains_(index)).unwrap_or(false);
 
-                // Skip empty cells
-                if cell.is_empty() && !selected {
-                    continue;
-                }
-
                 // Underline URL highlights
                 if self.url_highlight.as_ref().map(|range| range.contains_(index)).unwrap_or(false)
                 {
                     cell.inner.flags.insert(Flags::UNDERLINE);
+                } else if cell.is_empty() && !selected {
+                    continue;
                 }
```

Comment:
I feel like putting the code in the `else` block here is more confusing than helpful. While it does improve performance (unless LLVM is smart enough to optimize it itself, which I wouldn't be surprised about), I feel like it harms readability. And the performance difference will never show up in any benchmarks, because the url highlighting is not in any hot path.

So I'd propose something more like this:

```rust
if !cell.is_empty() || selected {
    return Some(RenderableCell::new(self.config, self.colors, cell, selected));
}
```

Since `cell.is_empty()` checks if the underline flag is set, this should not affect functionality, but would hopefully make it extremely clear that this conditional is not related to url highlighting at all, but just skips all empty cells.

Reply:
Cool, done!